src/GitHub.Services.Vssdk/Services/TippingService.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Windows;
+using GitHub.Logging;
+using Microsoft;
+using Serilog;
+using IServiceProvider = System.IServiceProvider;
+
+namespace GitHub.Services.Vssdk.Services
+{
+    /// <summary>
+    /// This service is a thin wrapper around <see cref="Microsoft.Internal.VisualStudio.Shell.Interop.IVsTippingService"/>.
+    /// </summary>
+    /// <remarks>
+    /// The <see cref="IVsTippingService"/> interface is public, but contained within the 'Microsoft.VisualStudio.Shell.UI.Internal' assembly.
+    /// To avoid a direct dependency on 'Microsoft.VisualStudio.Shell.UI.Internal', we use reflection to call this service.
+    /// </remarks>
+    public class TippingService : ITippingService
+    {
+        static readonly ILogger log = LogManager.ForContext<TippingService>();
+
+        // This is the only supported ClientId
+        public static readonly Guid ClientId = new Guid("D5D3B674-05BB-4942-B8EC-C3D13B5BD6EE");
+
+        readonly IServiceProvider serviceProvider;
+
+        public TippingService(IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+        }
+
+        /// <inheritdoc/>
+        public void RequestCalloutDisplay(Guid calloutId, string title, string message,
+            bool isPermanentlyDismissible, FrameworkElement targetElement,
+            Guid vsCommandGroupId, uint vsCommandId)
+        {
+            var screenPoint = !Splat.ModeDetector.InUnitTestRunner() ?
+                    targetElement.PointToScreen(new Point(targetElement.ActualWidth / 2, 0)) : default;
+            var point = new Microsoft.VisualStudio.OLE.Interop.POINT { x = (int)screenPoint.X, y = (int)screenPoint.Y };
+            RequestCalloutDisplay(ClientId, calloutId, title, message, isPermanentlyDismissible,
+                point, vsCommandGroupId, vsCommandId);
+        }
+
+        // The option to pass a command option is only available on Visual Studio 2017+.
+        void RequestCalloutDisplay(Guid clientId, Guid calloutId, string title, string message,
+            bool isPermanentlyDismissible, FrameworkElement targetElement,
+            Guid vsCommandGroupId, uint vsCommandId, object commandOption = null)
+        {
+            var tippingService = serviceProvider.GetService(typeof(SVsTippingService));
+            Assumes.Present(tippingService);
+            var currentMethod = MethodBase.GetCurrentMethod();
+            var parameterTypes = currentMethod.GetParameters().Select(p => p.ParameterType).ToArray();
+            var method = tippingService.GetType().GetMethod(currentMethod.Name, parameterTypes);
```

Comment:
I think I'd specify `BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly` ([REFL008](https://github.com/DotNetAnalyzers/ReflectionAnalyzers/blob/master/documentation/REFL008.md)).

Reply:
Ah, not `DeclaredOnly` since you're using `GetType()` instead of `typeof(SVsTippingService)`.

Would it possibly be safer to make an interface call (`typeof(IVsTippingService).GetMethod(...).Invoke(...)`) rather than duck-typing on whatever derived type is returned?

Reply:
Yes. What I meant by interface call was `typeof(IVsTippingService).GetMethod("RequestCalloutDisplay").Invoke(...)` which depends on a (hopefully stable) interface definition, instead of `tippingService.GetType().GetMethod("RequestCalloutDisplay").Invoke(...)` which does a duck-typing call on the (potentially internal and unstable) derived class.

Reply:
I think this is a new rule of thumb, that `.GetType().GetFoo(...)` really leads to pitfalls. I've seen this a few times now.  So far I can't think of a scenario when `typeof(AccessibleType)` or `assembly.GetType("InaccessibleType", throwOnError: ...)` wouldn't be better.

Reply:
Thanks for the warning. I'm now finding the interface and calling using that. 👍